app/src/main/java/org/wikipedia/views/ImagePreviewDialog.kt

```diff
@@ -104,39 +103,56 @@ class ImagePreviewDialog : ExtendedBottomSheetDialogFragment(), DialogInterface.
-                    else suggestedEditsSummary.pageTitle.description, false)
+                    SpannableString(if (suggestedEditsSummary.pageTitle.description.isNullOrEmpty()) suggestedEditsSummary.description
+                    else suggestedEditsSummary.pageTitle.description), null)
         }
-        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_artist), suggestedEditsSummary.metadata!!.artist(), false)
-        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_date), suggestedEditsSummary.metadata!!.dateTime(), false)
-        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_source), suggestedEditsSummary.metadata!!.imageDescriptionSource(), true)
-        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_licensing), suggestedEditsSummary.metadata!!.licenseShortName(), true)
+        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_artist), StringUtil.fromHtml(suggestedEditsSummary.metadata!!.artistUrl()), null)
+        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_date), StringUtil.fromHtml(suggestedEditsSummary.metadata!!.dateTime()), null)
+        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_source), StringUtil.fromHtml(suggestedEditsSummary.metadata!!.creditUrl()), null)
+        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_licensing), SpannableString(suggestedEditsSummary.metadata!!.licenseShortName()), getClickListenerFor(suggestedEditsSummary.metadata!!.licenseUrl()))
+        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_more_info), SpannableString(getString(R.string.suggested_edits_image_preview_dialog_file_page_link_text)), getClickListenerFor(getString(R.string.suggested_edits_image_file_page_commons_link, suggestedEditsSummary.title)))
         detailsHolder.requestLayout()
     }
 
-    private fun addDetailPortion(titleString: String, @Nullable detail: String?, shouldAddAccentTint: Boolean) {
-        if (!detail.isNullOrEmpty()) {
+    private fun addDetailPortion(titleString: String, @Nullable detail: Spanned, linkClickListener: View.OnClickListener?) {
+        if (!detail.isEmpty()) {
             val view = ImageDetailView(requireContext())
             view.titleTextView.text = titleString
-            if (shouldAddAccentTint) {
+            view.detailTextView.setMovementMethod(movementMethod)
+            if (linkClickListener != null) {
                 view.detailTextView.setTextColor(ResourceUtil.getThemedColor(context!!, R.attr.colorAccent))
+                view.externalLinkView.visibility = VISIBLE
+                view.detailsContainer.setOnClickListener(linkClickListener)
             }
-            view.detailTextView.text = StringUtil.strip(StringUtil.removeHTMLTags(detail))
+            view.detailTextView.text = detail
```

Comment:
Why don't just use the following:
```
view.detailTextView.text = StringUtil.strip(StringUtil.fromHtml(detail))
```

Reply:
![Screenshot_1562622962](https://user-images.githubusercontent.com/2435576/60846173-782d2600-a193-11e9-98f2-36b99abbccbc.png)